src/main/java/org/jfrog/hudson/pipeline/action/DeployedArtifactsAction.java

```diff
@@ -0,0 +1,56 @@
+package org.jfrog.hudson.pipeline.action;
+
+import hudson.model.Run;
+import jenkins.model.RunAction2;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Adds a list of the deployed artifacts to the summary of a pipeline job.
+ */
+public abstract class DeployedArtifactsAction implements RunAction2 {
+    private final List<DeployedArtifact> deployedArtifacts = new CopyOnWriteArrayList<>();
+
+    private Run build;
```

Comment:
Let's get rid of all warnings
```suggestion
    private Run<?, ?> build;
```